### PR TITLE
ETH2 Types: Epoch

### PR DESCRIPTION
**Pull request by farazdagi**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**
- Enable `types.Epoch`.
- Updates dependencies:
  - Runs on https://github.com/prysmaticlabs/ethereumapis/tree/feature/eth2-types

**Which issues(s) does this PR fix?**

Part of #8205

**Other notes for review**

### CODE REVIEW

**Review by terencechain — DISMISSED** [review]

**Review by rauljordan — DISMISSED** [review]
I went through every single file carefully and this PR is good to go, did not find anything contentious

**Comment by rkapka** [comment]
# :scream: 

https://github.com/prysmaticlabs/prysm/blob/f9303ca2e4e9f46fcfaf39b55feae5f155e7584d/endtoend/helpers/epochTimer.go#L55

**Comment by farazdagi** [comment]
@rkapka 
> prysm/endtoend/helpers/epochTimer.go

That's a misnomer, what is there is actually a slot number (I've that as `type.Slot` in my original custom types PR, and will update that file in the next `Slot` related PR). Basically, we are just ticking slots, but call them epoch ticker.

File: shared/sszutil/deep_equal.go [hunk 1]
```diff
@@ -115,6 +117,12 @@ func deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool, depth int) boo
 		}
 		return true
 	case reflect.Uint64:
+		switch v1.Type().Name() {
+		case "Epoch":
+			return v1.Interface().(types.Epoch) == v2.Interface().(types.Epoch)
```
Comment (nisdas) [inline shared/sszutil/deep_equal.go:122]: For deep value assertions to work shouldn't `uint64` and `types.Epoch` be equal ? This makes it only equal if they are of `types.Epoch`

Comment (farazdagi) [reply shared/sszutil/deep_equal.go:122]: Well, if we are comparing epochs, both should be epochs, imo. While underlying type of Epoch is uint64, and their integral values are the same, I believe if sth is epoch, it should be typed correctly, or not be considered as equal.

Comment (nisdas) [reply shared/sszutil/deep_equal.go:122]: Fair enough, the deep equal is only used in our pending queue while comparing the same type, so not a big issue. Its fine in that case.

**Comment by farazdagi** [comment]
Note: Per Nishant's request will sync from genesis and run some validators on top of this branch. If everything works, will merge this PR.

**Review by nisdas — APPROVED** [review]

**Pull request merged**